### PR TITLE
Refactor: all components attributes are reordered

### PR DESCRIPTION
**Pull request by luuu-xu**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Went through all components and pages to make sure that now components and HTML elements have their attributes ordered like this:

```
<element
  conditionals=""
  class=""
  :class="{}"
  props=""
  aria-label=""
></element>
```

### Related issue

- #339

### CODE REVIEW

File: frontend/components/card/CardConnect.vue [hunk 1]
```diff
@@ -9,21 +9,19 @@
       >
         <Icon
           v-if="userIsAdmin && !editModeEnabled"
-          @click="toggleEditMode"
           name="bi:pencil-square"
           size="1.2em"
+          @click="toggleEditMode"
```
Comment (andrewtavis) [inline frontend/components/card/CardConnect.vue:14]: I was thinking that the `@` attributes could be below conditionals? Was something that came to mind when looking at things last night.

Comment (andrewtavis) [reply frontend/components/card/CardConnect.vue:14]: I can go through and change these with #151?

File: frontend/components/card/CardDangerZone.vue [hunk 2]
```diff
@@ -3,50 +3,44 @@
     class="px-5 py-5 border bg-light-act-red/30 dark:bg-dark-act-red/30 border-light-act-red dark:border-dark-act-red card-style-base text-light-text dark:text-dark-text"
   >
     <div class="flex flex-col gap-5">
-      <h3
-        class="font-bold text-left text-light-act-red dark:text-dark-act-red responsive-h3 font-display"
-      >
+      <h3 class="font-bold text-left text-light-act-red dark:text-dark-act-red responsive-h3 font-display">
         {{ $t("components.card-danger-zone.header") }}
       </h3>
-      <p class="">{{ description }}</p>
```
Comment (andrewtavis) [inline frontend/components/card/CardDangerZone.vue:11]: Where did all these come from?? 🤦‍♂️😅

File: frontend/components/card/CardDangerZone.vue [hunk 3]
```diff
@@ -3,50 +3,44 @@
...
           </div>
           <div class="flex flex-col gap-2">
             <label
-              for="password"
               class="font-bold text-light-text dark:text-dark-text responsive-h4"
-              :placeholder="
-                $t('components.card-danger-zone.placeholder-password')
-              "
-              >{{ $t("components.card-danger-zone.label-password") }} *</label
+              for="password"
```
Comment (andrewtavis) [inline frontend/components/card/CardDangerZone.vue:28]: This prop should likely be renamed 🤔

File: frontend/components/card/search-result/CardSearchResultResource.vue [hunk 4]
```diff
@@ -43,9 +43,7 @@
           <MetaTagLocation :location="resource?.relatedLocation" />
           <MetaTagDate :date="resource.creationDate.toLocaleDateString()" />
         </div>
-        <div
-          class="flex justify-center space-x-3 lg:space-x-4 md:justify-start"
-        >
+        <div class="flex justify-center space-x-3 lg:space-x-4 md:justify-start">
```
Comment (andrewtavis) [inline frontend/components/card/search-result/CardSearchResultResource.vue:46]: Something seems to be wrong with this here as it's not formatted as a string. Is problematic before, so just wanted to bring it up so we can fix it 😊

File: frontend/components/card/search-result/CardSearchResultUser.vue [hunk 5]
```diff
@@ -23,21 +17,15 @@
...
-        <div
-          class="flex items-center justify-center w-full md:hidden space-x-3"
-        >
+        <div class="flex items-center justify-center w-full md:hidden space-x-3">
           <MetaTagLocation :location="user?.location" />
         </div>
-        <div
-          class="flex justify-center space-x-3 lg:space-x-4 md:justify-start"
-        >
+        <div class="flex justify-center space-x-3 lg:space-x-4 md:justify-start">
```
Comment (andrewtavis) [inline frontend/components/card/search-result/CardSearchResultUser.vue:28]: Similar note on the formatting of the class here :) Seems to be problematic given it's not showing up as a string :)

File: frontend/components/feed/FeedItem.vue [hunk 6]
```diff
@@ -27,7 +31,7 @@
     <div
       class="w-full h-32 rounded-md bg-light-placeholder dark:bg-dark-placeholder"
     >
-      <img :src="item.imgURL" alt="" class="" />
+      <img class="" :src="item.imgURL" alt="" />
```
Comment (andrewtavis) [inline frontend/components/feed/FeedItem.vue:34]: You meant to delete this one?

File: frontend/components/form/FormButtonStyledCheckbox.vue [hunk 7]
```diff
@@ -7,16 +7,16 @@
...
       v-bind="{ ...$attrs, onChange: updateValue }"
       :checked="modelValue"
       :id="uuid"
-      class="hidden"
     />
     <label
       v-if="label"
-      :for="uuid"
       class="min-h-[2.75rem] absolute w-full h-full flex items-center justify-center cursor-pointer"
+      :for="uuid"
```
Comment (andrewtavis) [inline frontend/components/form/FormButtonStyledCheckbox.vue:19]: Noting that this should also be changed to be more descriptive :)

File: frontend/components/form/FormButtonStyledRadio.vue [hunk 8]
```diff
@@ -4,20 +4,20 @@
...
+      class="hidden"
       type="radio"
       v-bind="{ ...$attrs, onChange: updateValue }"
       :checked="modelValue === value"
       :id="uuid"
-      class="hidden"
     />
     <label
+      class="min-h-[2.75rem] absolute w-full h-full flex items-center justify-center cursor-pointer"
       v-if="label"
```
Comment (andrewtavis) [inline frontend/components/form/FormButtonStyledRadio.vue:19]: `v-if` should be above :)

File: frontend/components/form/FormCheckbox.vue [hunk 9]
```diff
@@ -1,11 +1,11 @@
 <template>
   <div class="relative flex flex-row items-center justify-start mb-3">
     <input
+      class="cursor-pointer mb-0 peer appearance-none w-[1.375rem] h-[1.375rem] border border-black rounded-sm bg-light-button dark:bg-dark-button dark:border-dark-text"
       type="checkbox"
       v-bind="{ ...$attrs, onChange: updateValue }"
```
Comment (andrewtavis) [inline frontend/components/form/FormCheckbox.vue:6]: Should `v-bind` also be above. I don't think we'll have many instances of multiple `v-` attributes.

File: frontend/components/sidebar/left/SidebarLeftIndex.vue [hunk 10]
```diff
@@ -55,7 +51,10 @@
             :active="button.active"
           />
         </li>
-        <li v-if="pageType === 'event'" v-for="button in eventButtons">
+        <li
+          v-if="pageType === 'event'"
```
Comment (andrewtavis) [inline frontend/components/sidebar/left/SidebarLeftIndex.vue:55]: Note: based on the [Vue styleguide](https://vuejs.org/style-guide/) this should be split. We should have a parent `v-if` `<div>`. Was reading through it the other day :)

**Review by andrewtavis — CHANGES_REQUESTED** [review]
Requesting changes so we have a finalized point where the review's done, @luuu-xu :) Big thing is to check with you on where you think the `@` attributes should go. I updated the [Vue/Nuxt section in the style guide](https://github.com/activist-org/activist/blob/main/STYLEGUIDE.md#vue-and-nuxt-) with a snippet for how attributes should be ordered to reflect this. Aside from that there are a few minor changes :)

Thanks @luuu-xu! We're almost to a awesome basis for frontend code quality!! 🚀

File: frontend/components/card/search-result/CardSearchResultResource.vue (hunk 4 above)
Comment (andrewtavis) [reply frontend/components/card/search-result/CardSearchResultResource.vue:46]: Nvm, I guess GitHub formatting is hallucinating 🤷‍♂️

File: frontend/components/card/CardConnect.vue (hunk 1 above)
Comment (luuu-xu) [reply frontend/components/card/CardConnect.vue:14]: What do you think? I don't have very much experience with Vue projects so I am not sure. To my guts, I think putting `@` attributes at the very bottom seems fine.

File: frontend/components/card/CardDangerZone.vue (hunk 2 above)
Comment (luuu-xu) [reply frontend/components/card/CardDangerZone.vue:11]: You mean the empty `class` attributes?

File: frontend/components/card/CardDangerZone.vue (hunk 3 above)
Comment (luuu-xu) [reply frontend/components/card/CardDangerZone.vue:28]: Which prop do you mean that needs to be renamed?

File: frontend/components/card/CardConnect.vue (hunk 1 above)
Comment (andrewtavis) [reply frontend/components/card/CardConnect.vue:14]: We talked it over in the dev sync and folks were generally on board with having `@` attributes towards the top so the component's usage is prioritized and so we have only one focus point of the `aria-label` at the end :)

File: frontend/components/card/CardDangerZone.vue (hunk 2 above)
Comment (andrewtavis) [reply frontend/components/card/CardDangerZone.vue:11]: Yes :) No stress on this, just a bit of light venting 😊

File: frontend/components/card/CardDangerZone.vue (hunk 3 above)
Comment (andrewtavis) [reply frontend/components/card/CardDangerZone.vue:28]: Nevermind on this, @luuu-xu! In the review I got a bit confused at this point and I was thinking that `for` was a prop on a component. A `for` attribute on a `<label>` is totally fine :)

**Comment by andrewtavis** [comment]
I think that this should be good to merge post removing the `class=""` mentioned above, @luuu-xu :) If you want to put the `@attributes` after `v-attributes` that'd be fine, or I can also get to them in #151's final sweep :)

**Comment by andrewtavis** [comment]
I'll go through and do the final changes for here with what I'll do for #151, @luuu-xu :) Thanks for all the changes here!

**Comment by luuu-xu** [comment]
@andrewtavis Great! Thank you for the additional pair of hands on these two issues. I will go ahead and pull from your commit and start working on other issues now.

**Comment by andrewtavis** [comment]
Looking forward to the future commits, @luuu-xu! :) :)

**Pull request merged**